nat-lab/tests/test_mesh_connected_socket.py

```diff
@@ -0,0 +1,103 @@
+import asyncio
+import pytest
+import telio
+from contextlib import AsyncExitStack
+from helpers import SetupParameters, setup_mesh_nodes
+from telio import PathType, State
+from telio_features import TelioFeatures, Direct
+from utils.connection_util import ConnectionTag
+from utils.iperf3 import IperfServer, IperfClient, Protocol
+from utils.ping import Ping
+
+
+# Marks in-tunnel stack only, exiting only through IPv4
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "disable_connected_socket_a",
+    [True, False],
+)
+@pytest.mark.parametrize(
+    "disable_connected_socket_b",
+    [True, False],
+)
+async def test_mesh_connected_socket(
+    disable_connected_socket_a, disable_connected_socket_b
+) -> None:
+    async with AsyncExitStack() as exit_stack:
+        features_alpha = TelioFeatures(
+            direct=Direct(providers=["stun", "local"]),
+            disable_connected_socket=disable_connected_socket_a,
+        )
+        features_beta = TelioFeatures(
+            direct=Direct(providers=["stun", "local"]),
+            disable_connected_socket=disable_connected_socket_b,
+        )
+        env = await setup_mesh_nodes(
+            exit_stack,
+            [
+                SetupParameters(
+                    connection_tag=ConnectionTag.DOCKER_CONE_CLIENT_1,
+                    adapter_type=telio.AdapterType.BoringTun,
+                    features=features_alpha,
+                ),
+                SetupParameters(
+                    connection_tag=ConnectionTag.DOCKER_CONE_CLIENT_2,
+                    adapter_type=telio.AdapterType.BoringTun,
+                    features=features_beta,
+                ),
```

Comment:
The PR enables connected sockets on apple platforms - in that case it would be more appropriate to use mac vm instead of linux docker as node(s).

Reply:
Good point, we only have one VM though, but still would be good to test the combination.